SERVER/src/database.py

```
@@ -1,99 +1,252 @@
+        query = "INSERT INTO training_data (Actor_id, input_text, output_text) VALUES (?, ?, ?)"
+        self._execute_query(query, (Actor_id, dataset["input"], dataset["output"]), commit=True)
+
+    def get_training_data_for_Actor(self, Actor_id):
+        rows = self._execute_query("SELECT input_text, output_text FROM training_data WHERE Actor_id=?", (Actor_id,), fetchall=True)
+        return [{"input": row["input_text"], "output": row["output_text"]} for row in rows] if rows else []
+
+    def save_client_token(self, Actor_id, token):
+        """Saves a new token, defaults status to 'Registered'."""
+        # Ensure character exists or create a placeholder if it's Actor1
+        if Actor_id == "Actor1" and not self.get_character("Actor1"):
+            self.save_character(name="ServerChar", personality="Host", goals="Manage story", backstory="Server's own character",
```

Pylint (3.3.7)

[convention] Line too long (128/100)

(C0301)